devtools/maven/src/main/java/io/quarkus/maven/NativeImageAgentMojo.java

```diff
@@ -0,0 +1,123 @@
+package io.quarkus.maven;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.Arrays;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+
+import io.quarkus.builder.Json;
+import io.quarkus.builder.JsonReader;
+import io.quarkus.builder.JsonTransform;
+import io.quarkus.builder.json.JsonMember;
+import io.quarkus.builder.json.JsonObject;
+import io.quarkus.builder.json.JsonString;
+import io.quarkus.builder.json.JsonValue;
+
+@Mojo(name = "native-image-agent", defaultPhase = LifecyclePhase.POST_INTEGRATION_TEST, requiresDependencyResolution = ResolutionScope.COMPILE_PLUS_RUNTIME, threadSafe = true)
+public class NativeImageAgentMojo extends QuarkusBootstrapMojo {
```

Comment:
@galderz could you please add a class-level Javadoc here? It will be displayed in the description of this goal by Maven then.